src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.DivRem.cs

```diff
@@ -0,0 +1,387 @@
+                    digit = 0xFFFFFFFF;
+
+                // Our first guess may be a little bit to big
+                ulong check = divHi * digit + ((divLo * digit) >> 32);
+                if (check > valHi)
+                    --digit;
+
+                // Our guess may be still a little bit to big
+                do
+                {
+                    MultiplyDivisor(divisor, divisorLength, digit, guess);
+                    guessLength = guess[divisorLength] == 0
+                                ? divisorLength : divisorLength + 1;
+                    delta = Compare(dividend + n, guessLength,
+                                    guess, guessLength);
+                    if (delta < 0)
+                        --digit;
+                }
+                while (delta < 0);
+
+                // We have the digit!
+                SubtractSelf(dividend + n, guessLength,
+                             guess, guessLength);
+                quotient[n] = (uint)digit;
+            }
+
+            if (remainderLength != 0)
+            {
+                // Repairing the remaining dividend gets the remainder
+                RightShift(dividend, divisorLength,
+                           remainder, remainderLength,
+                           shift);
+            }
+        }
+
+        [SecuritySafeCritical]
+        private unsafe static void LeftShift(uint* value, int valueLength,
+                                             uint* target, int targetLength,
+                                             int shift)
+        {
+            Debug.Assert(valueLength >= 1);
+            Debug.Assert(targetLength == valueLength ||
+                         targetLength == valueLength + 1);
+            Debug.Assert(shift >= 0 && shift < 32);
+
+            if (shift > 0)
+            {
+                int backShift = 32 - shift;
+                target[0] = value[0] << shift;
+                for (int i = 1; i < valueLength; i++)
+                    target[i] = (value[i] << shift)
+                        | (value[i - 1] >> backShift);
+                if (targetLength > valueLength)
+                    target[valueLength] =
+                        value[valueLength - 1] >> backShift;
+            }
+            else
+            {
+                for (int i = 0; i < valueLength; i++)
+                    target[i] = value[i];
+            }
+        }
+
+        [SecuritySafeCritical]
+        private unsafe static void RightShift(uint* value, int valueLength,
+                                              uint* target, int targetLength,
+                                              int shift)
+        {
+            Debug.Assert(valueLength >= 1);
+            Debug.Assert(targetLength == valueLength);
+            Debug.Assert(shift >= 0 && shift < 32);
+
+            if (shift > 0)
+            {
+                int backShift = 32 - shift;
+                for (int i = 0; i < valueLength - 1; i++)
```

Comment:
Per the style guide, can you please add curly braces around multi-line loop blocks?